### PR TITLE
[KBFS-2208] block_retrieval_worker_test: Add logging to track down flake

### PR DESCRIPTION
**Pull request by jzila**

### CODE REVIEW

File: libkbfs/block_retrieval_worker_test.go [hunk 1]
```diff
@@ -263,7 +263,9 @@ func TestBlockRetrievalWorkerShutdown(t *testing.T) {
 	shutdown := false
 	select {
 	case <-ch:
+		t.Fatal("Expected not to retrieve a result from the Request.")
```
Comment (songgao) [inline libkbfs/block_retrieval_worker_test.go]: This doesn't fix the flake does it? It still flakes the same way I think?

**Comment by songgao** [comment]
Also I was a bit confused by https://github.com/keybase/kbfs/pull/1030/files#diff-e3ef24ce4d67246c2920d7d276db5648R273 and checking the `shutdown` variable again. The variable doesn't change after that call

File: libkbfs/block_retrieval_worker_test.go (hunk 1 above)
Comment (jzila) [inline libkbfs/block_retrieval_worker_test.go]: Yeah but at least we know which scenario is happening.

**Comment by jzila** [comment]
Yeah that check doesn't really do anything.

File: libkbfs/block_retrieval_worker_test.go (hunk 1 above)
Comment (songgao) [inline libkbfs/block_retrieval_worker_test.go]: Ah got you. I didn't read the PR title fully

File: libkbfs/block_retrieval_worker_test.go [hunk 2]
```diff
@@ -263,13 +263,26 @@ func TestBlockRetrievalWorkerShutdown(t *testing.T) {
...
 	case continueCh <- nil:
+		t.Fatal("Expected the block getter not to be receiving.")
 	default:
 		shutdown = true
 	}
 	require.True(t, shutdown)
-	w.Shutdown()
-	require.True(t, shutdown)
+
+	// Ensure the test completes in a reasonable time.
```
Comment (jzila) [inline libkbfs/block_retrieval_worker_test.go]: @songgao: I replaced the last check with the right one.

Comment (songgao) [inline libkbfs/block_retrieval_worker_test.go]: nice!

File: libkbfs/block_retrieval_worker_test.go [hunk 3]
```diff
@@ -263,13 +263,26 @@ func TestBlockRetrievalWorkerShutdown(t *testing.T) {
...
+	timer := time.NewTimer(10 * time.Second)
+	doneCh := make(chan struct{})
+	go func() {
+		w.Shutdown()
```
Comment (songgao) [inline libkbfs/block_retrieval_worker_test.go]: Is `w.Shutdown()` synchronous? https://github.com/keybase/kbfs/pull/1030/files#diff-e3ef24ce4d67246c2920d7d276db5648R264 seems to suggest it's async?

Comment (jzila) [inline libkbfs/block_retrieval_worker_test.go]: Yes. This tests that.

Comment (jzila) [inline libkbfs/block_retrieval_worker_test.go]: Hence the `t.Fatal("Expected another Shutdown not to block.")`

**Pull request merged**